### PR TITLE
PT-168026587 GA in dry run

### PR DESCRIPTION
**Pull request by hanssv**
Improve dry-run - handle GA transactions, etc

- Add GAAttachTx
- Allow normal (unsigned) transactions from generalized accounts
- Add DryRunCallRequest - where you can pass a subset of contract call parameters and set the context (i.e. Auth context).

### CODE REVIEW

File: py/hack_dry_run_input.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
+# coding: utf-8
+
```
Comment (dincho) [inline py/hack_dry_run_input.py]: 😢

File: apps/aecore/src/aec_dry_run.erl [hunk 2]
```diff
@@ -64,8 +78,65 @@ add_account(#{pub_key := PK, amount := A}, AccountsTree) ->
...
+        Context    = TxHashCtxt ++ StateCtxt,
+        {tx, Context, dummy_sign(CallTx)}
+    catch _:_R ->
+        error({bad_dry_run_call_request, ReqMap})
+    end.
+
+prepare_env(Env, Opts) ->
+    case proplists:get_value(auth_tx, Opts, undefined) of
+        undefined -> Env;
+        TxHash    -> aetx_env:set_ga_tx_hash(Env, TxHash)
```
Comment (ThomasArts) [inline apps/aecore/src/aec_dry_run.erl]: Would it add value to check that TxHash is a binary hash?

File: apps/aeprimop/src/aeprimop.erl [hunk 3]
```diff
@@ -507,6 +509,12 @@ inc_account_nonce({Pubkey, Nonce, Force}, #state{} = S) ->
                 _ ->
                     Account1 = aec_accounts:set_nonce(Account, Nonce),
                     put_account(Account1, S1)
+            end;
+        false when DryRun andalso AccType == generalized ->
```
Comment (ThomasArts) [inline apps/aeprimop/src/aeprimop.erl]: What if we are in Dryrun and AccType == basic?

Comment (hanssv) [reply apps/aeprimop/src/aeprimop.erl]: Then we do the same checks as if we are not dry-running.

File: apps/aecore/src/aec_dry_run.erl (hunk 2 above)
Comment (hanssv) [reply apps/aecore/src/aec_dry_run.erl]: It comes from `aeser_api_encoder:safe_decode/2` so seems like overkill

**Pull request merged**